examples/rwalk_mnist.py

```diff
@@ -0,0 +1,149 @@
+import torch
+from os.path import expanduser
+import argparse
+from torchvision.datasets import MNIST
+from torchvision.transforms import ToTensor
+from avalanche.benchmarks import PermutedMNIST, nc_benchmark
+from avalanche.training.supervised import RWalk
+from avalanche.models import SimpleMLP
+from avalanche.evaluation.metrics import (
+    forgetting_metrics,
+    accuracy_metrics,
+    loss_metrics,
+    bwt_metrics,
+)
+from avalanche.logging import InteractiveLogger, TensorboardLogger
+from avalanche.training.plugins import EvaluationPlugin
+
+"""
+This example tests RWalk on Split MNIST and Permuted MNIST.
+It is possible to choose, among other options, between EWC with separate
+penalties and online EWC with a single penalty.
+"""
+
+
+def main(args):
+    model = SimpleMLP(hidden_size=args.hs)
+    optimizer = torch.optim.Adam(model.parameters(), lr=args.lr)
+    criterion = torch.nn.CrossEntropyLoss()
+
+    # check if selected GPU is available or use CPU
+    assert args.cuda == -1 or args.cuda >= 0, "cuda must be -1 or >= 0."
+    device = torch.device(
+        f"cuda:{args.cuda}"
+        if torch.cuda.is_available() and args.cuda >= 0
+        else "cpu"
+    )
+    print(f"Using device: {device}")
+
+    # create scenario
+    if args.scenario == "pmnist":
+        scenario = PermutedMNIST(n_experiences=args.permutations)
+    elif args.scenario == "smnist":
+        mnist_train = MNIST(
+            root=expanduser("~") + "/.avalanche/data/mnist/",
+            train=True,
+            download=True,
+            transform=ToTensor(),
+        )
+        mnist_test = MNIST(
+            root=expanduser("~") + "/.avalanche/data/mnist/",
+            train=False,
+            download=True,
+            transform=ToTensor(),
+        )
+        scenario = nc_benchmark(
+            mnist_train, mnist_test, 5, task_labels=False, seed=1234
+        )
+    else:
+        raise ValueError("Wrong scenario name. Allowed pmnist, smnist.")
+
+    # choose some metrics and evaluation method
+    interactive_logger = InteractiveLogger()
+    tensorboard_logger = TensorboardLogger()
+    eval_plugin = EvaluationPlugin(
+        accuracy_metrics(
+            minibatch=True, epoch=True, experience=True, stream=True
+        ),
+        loss_metrics(minibatch=True, epoch=True, experience=True, stream=True),
+        forgetting_metrics(experience=True, stream=True),
+        bwt_metrics(experience=True, stream=True),
+        loggers=[interactive_logger, tensorboard_logger],
+    )
+
+    # create strategy
+    strategy = RWalk(
```

Comment:
if you remove the class, this becomes a Naive with RWalkPlugin

Reply:
Thanks, maybe I'll add an option to use also a replay plugin.

Reply:
If you remove the strategy I think you can just use Naive + Rwalk plugin